### PR TITLE
[python-package] add install option to enable printing of time costs

### PR DESCRIPTION
**Pull request by Remy-Luciani**
According to the [install guide](https://lightgbm.readthedocs.io/en/latest/Installation-Guide.html):
> Users who want to perform benchmarking can make LightGBM output time costs for different internal routines by adding -DUSE_TIMETAG=ON to CMake flags.

This feature was unfortunately not available when installing LightGBM from the Python package. This PR attempts to fix this. :)

### CODE REVIEW

File: python-package/setup.py
Comment (jmoralez) [inline python-package/setup.py:159]:
```suggestion
    if time_costs:
```

Comment (Remy-Luciani) [reply python-package/setup.py]: Oh my, sorry for the silly mistakes...

File: python-package/setup.py
Comment (jmoralez) [inline python-package/setup.py:294]:
```suggestion
        self.time_costs = False
```

Comment (Remy-Luciani) [reply python-package/setup.py]: My apologies. 🤦

Comment (jmoralez) [reply python-package/setup.py]: Haha no worries. Thank you for your contribution!

**Comment by jmoralez** [comment]
@jameslamb the dask tests seem to be getting stuck and we get timeouts on the jobs, ([Linux regular this PR](https://dev.azure.com/lightgbm-ci/lightgbm-ci/_build/results?buildId=13611&view=logs&j=c28dceab-947a-5848-c21f-eef3695e5f11), [several Linux jobs in another PR](https://dev.azure.com/lightgbm-ci/lightgbm-ci/_build/results?buildId=13596&view=results). Should we open an issue for this? I'll try to replicate the problem later.

**Comment by Remy-Luciani** [comment]
@jmoralez Could I do something to help? I can re-push something to launch the CI jobs again but maybe some problems are remaining? Let me know :)

**Comment by jameslamb** [comment]
@Remy-Luciani LightGBM's CI is facing some significant issues right now (see #5509 and #5510), and we won't be able to come back to this PR until they've been addressed. Sorry for the inconvenience.

**Comment by jameslamb** [comment]
@jmoralez to answer your question directly.... yes I think we should open a separate issue for the Dask timeout thing, and I'd be grateful if you have time to investigate it. (https://github.com/microsoft/LightGBM/pull/5510#issuecomment-1260363349)

**Comment by jameslamb** [comment]
hey @Remy-Luciani , happy to say that we've fixed LightGBM's CI issues (#5510)! Thanks very much for your patience.

If you can please merge the latest changes on `master` into this branch, we'll review this again and try to move it forward.

File: python-package/README.rst
Comment (jameslamb) [inline python-package/README.rst:173]:
```suggestion
Use this option to make LightGBM output time costs for different internal routines, to investigate and benchmark its performance.
```

Will you please switch to this wording which is a bit more direct (e.g. "Use this..." instead of "Users who...")? It's a minor personal preference, but I prefer language like that for documentation like this.

**Review by jameslamb — APPROVED** [review]
One minor comment, otherwise this looks good to me and I support adding this option to the Python package.

But I'd also like @jmoralez or @StrikerRUS to approve before this is merged.

File: python-package/README.rst
Comment (Remy-Luciani) [reply python-package/README.rst:173]: Alright! FYI, I've mostly copy/pasted it from the [installation guide](https://lightgbm.readthedocs.io/en/latest/Installation-Guide.html) to remain coherent. In case you would like to change it there too.

**Comment by Remy-Luciani** [comment]
All feedbacks have been taken and the branch is up to date with master! :) Let me know if there are another feedbacks

**Review by jameslamb — APPROVED** [review]
thanks for the changes! looks good to me.

**Review by jmoralez — APPROVED** [review]

**Comment by jameslamb** [comment]
closing and re-opening to try to fix this flaky `license/cla` job (sorry about that!)

Screen Shot 2022-10-11 at 11 13 35 AM

That seemed to fix it last time we saw something similar: https://github.com/microsoft/LightGBM/pull/5160#issuecomment-1116327139

**Comment by Remy-Luciani** [comment]
@microsoft-github-policy-service agree [company="Theodo"]

**Comment by jameslamb** [comment]
I'm so sorry @Remy-Luciani . These comments and commands look new.

The CLA bot is something controlled by Microsoft at an organization level, not something that we (the maintainers of LightGBM) control or configure 😬 .

So that's bot's comments are as confusing to me as they probably are to you 😭

**Comment by Remy-Luciani** [comment]
> I'm so sorry @Remy-Luciani . These comments and commands look new.
> 
> The CLA bot is something controlled by Microsoft at an organization level, not something that we (the maintainers of LightGBM) control or configure 😬 .
> 
> So that's bot's comments are as confusing to me as they probably are to you 😭

No worries. I don't know if you can make some feedback to someone at Microsoft since the first message was misleading by adding brackets to the command syntax. 😛

**Comment by Remy-Luciani** [comment]
@microsoft-github-policy-service agree company="Theodo"

**Pull request merged**
